### PR TITLE
BIGC-711 Web register on cornerstone and fortune themes.

### PR DESCRIPTION
**Pull request by sofib**

### CODE REVIEW

File: src/modules/app_commands/sendPipelineRequest.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/**
+ * App command to send pipeline request.
+ *
+ * @param {string} pipelineName Name of the pipeline.
+ * @param {string} pipelineInput Input parameters for the pipeline.
```
Comment (Menes1337) [inline src/modules/app_commands/sendPipelineRequest.js]: should be type Object, right?

File: src/modules/app_commands/sendPipelineRequest.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      input: pipelineInput,
+    },
+  };
+}
+
+/**
+ * App command to send trusted pipeline request.
+ *
+ * @param {string} pipelineName Name of the pipeline.
+ * @param {string} pipelineInput Input parameters for the pipeline.
```
Comment (Menes1337) [inline src/modules/app_commands/sendPipelineRequest.js]: here also. You can also write the optional value (and the default value) in the jsdoc: [pipelineInput={}]

File: src/modules/cookies.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/**
+ * Returns value for the specified cookie.
+ *
+ * @param {string} cname Cookie name.
+ * @returns {string}
+ */
+export function getCookie(cname) {
```
Comment (Menes1337) [inline src/modules/cookies.js]: Just a suggestion to rename it to cookieName

File: src/modules/SGEvent.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { notifyEventReceived } from './appEvents';
+
+export default {
+  __call: function call(eventName, eventArguments) {
+    console.log(`${`# Received event ${eventName}`}$`);
```
Comment (Menes1337) [inline src/modules/SGEvent.js]: I guess you want to leave the console.log for reason (debugging reason in the future)

File: src/modules/app_commands/sendPipelineRequest.js (hunk 1 above)
Comment (sofib) [reply src/modules/app_commands/sendPipelineRequest.js]: Right, done.

File: src/modules/app_commands/sendPipelineRequest.js (hunk 2 above)
Comment (sofib) [reply src/modules/app_commands/sendPipelineRequest.js]: Done

File: src/modules/cookies.js (hunk 3 above)
Comment (sofib) [reply src/modules/cookies.js]: Done. It was a reuse of code, so I didn't pay attention to naming :)

File: src/modules/SGEvent.js (hunk 4 above)
Comment (sofib) [reply src/modules/SGEvent.js]: Yes, the same is in other extension we have, this is why I left it.

File: test/unit/modules/urlTest.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    it('should not find sgcloud_callback_data paramerter when it was not provided', () => {
+      let result = getParameterFromQueryString('?action=create_account', 'sgcloud_callback_data');
+      expect(result).equals(null);
+      result = getParameterFromQueryString('', 'sgcloud_callback_data');
+      expect(result).equals(null);
+      result = getParameterFromQueryString('?', 'sgcloud_callback_data');
+      expect(result).equals(null);
+    });
+
+    it('should find sgcloud_checkout', () => {
```
Comment (Menes1337) [inline test/unit/modules/urlTest.js]: Is this actually doing anything?

File: src/modules/SGEvent.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { notifyEventReceived } from './appEvents';
+
+export default {
+  __call: function call(eventName, eventArguments) {
```
Comment (Menes1337) [inline src/modules/SGEvent.js]: please add jsdoc

File: src/modules/appEvents.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  subscribers[eventName].push(subscriberCallback);
+}
+
+/**
+ * Notify all subscribed callback when an app event is received.
+ *
+ * @param {stirng} eventName Name of the app event.
+ * @param {function} eventArguments Arguments of the event.
```
Comment (Menes1337) [inline src/modules/appEvents.js]: The type is Array - at least its getting called from src/modules/appEvents.js with an Array

File: src/modules/app_commands/onLoad.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+/**
+ * App command onload#
```
Comment (Menes1337) [inline src/modules/app_commands/onLoad.js]: character "#" can be removed

File: test/unit/modules/urlTest.js (hunk 5 above)
Comment (sofib) [reply test/unit/modules/urlTest.js]: No, removed.

File: src/modules/SGEvent.js (hunk 6 above)
Comment (sofib) [reply src/modules/SGEvent.js]: done

File: src/modules/appEvents.js (hunk 7 above)
Comment (sofib) [reply src/modules/appEvents.js]: done

File: src/modules/app_commands/onLoad.js (hunk 8 above)
Comment (sofib) [reply src/modules/app_commands/onLoad.js]: done.

File: src/modules/shopgateApp.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
+import { getParameterFromQueryString } from './url';
+import { getCookie, setCookie } from './cookies';
+
+/**
+ * Returns query string parameter form url.
+ *
+ * @param {string} url An url.
+ * @returns {string}
+ */
+function getQSFromUrl(url) {
```
Comment (Menes1337) [inline src/modules/shopgateApp.js]: Is this function maybe better placed in url.js ?

**Review by Menes1337 — CHANGES_REQUESTED** [review]
please have a look at my comments

File: src/modules/shopgateApp.js (hunk 9 above)
Comment (sofib) [reply src/modules/shopgateApp.js]: done

**Review by Menes1337 — APPROVED** [review]

**Pull request merged**